### DISCUSSION TITLE
Proper types with implementing `ResolverInterface` and inheritance.

### DISCUSSION THREAD

**Discussion by valerii15298** [MichalLytek/type-graphql#1263] [Q&A] [answered]
I am trying to implement `FieldResolver` while using resolver for `ObjectType` like this:
```ts
@ObjectType()
class Type1 {
  @Field()
  field1: string;
}

@Resolver((_of) => Type1)
class Res1 implements ResolverInterface<Type1> {
  @FieldResolver()
  async field1(@Arg('arg1') arg1: string) {
    return 'hello';
  }
}
```
And the problem is that you cant use `Arg` decorator because of implementing `ResolverInterface<Type1>` interface.
Typescript complains:
```
TS2416: Property 'field1' in type 'Res1' is not assignable to the same property in base type 'ResolverInterface<Type1>'.
Type '(arg1: string) => Promise<string>' is not assignable to type '(root: Type1, ...args: any[]) => string | Promise<string>'.
Types of parameters 'arg1' and 'root' are incompatible.
Type 'Type1' is not assignable to type 'string'.
```
The same but without `@Arg` works fine:
```ts
@ObjectType()
class Type1 {
  @Field()
  field1: string;
}

@Resolver((_of) => Type1)
class Res1 implements ResolverInterface<Type1> {
  @FieldResolver()
  async field1() {
    return 'hello';
  }
}
```
Is there some way to handle it? I did not found examples in docs.
I can just simply remove part `implements ResolverInterface<Type1>` and everything will work fine,
but maybe there is better solution to benefit from type safety?

**Comment by MichalLytek** [comment, marked as answer]
In 90% of cases, field resolvers are making computation based on the parent object (`root` in the error message).
So you can:
1) Use `@Root() root: Type1` as the first parameter
2) Remove `ResolverInterface`
3) Write your own `ResolverInterface` type that does not check the first argument at all.

**Reply by valerii15298** [reply]
Thank you for such a quick answers always and clear concise explanation.
